### ISSUE TITLE
large file problem

### ISSUE THREAD

**Issue by vdhulst** [HERA-Team/aipy#31]
crash occurs when reading large miriad uv files. The manifestation of the problem from within python is as follows:
>>> import aipy
>>> uv=aipy.miriad.UV('NGC315.mir')
### Fatal Error:  Bad file descriptor
Segmentation fault (core dumped)

the data set can be obtained via:
"scp -r anonymous@ftp.astron.nl:/ftp/pub/export/astron/adebahr/aipy/NGC315.mir . "

tests on smaller files extracted from this data using uvcat indicate that this error does not occur for file sizes under 16GB (i.e. data sets under 9h extracted from this 12h observation do not give this error).

we are using aipy version 2.1.9

**Comment by pkgw** [comment]
What OS and CPU type? The problem might be platform-dependent.

**Comment by vdhulst** [comment]
uname -a provides:  4.4.0-109-generic #132-Ubuntu SMP Tue Jan 9 19:52:39 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

**Comment by pkgw** [comment]
And what is the password for that `scp` command? My guesses (email address, `anonymous`) aren't working.

**Comment by vdhulst** [comment]
cpu is AMD Opteron

**Comment by vdhulst** [comment]
will arrange to fix the scp link

**Comment by vdhulst** [comment]
@pkgw:   it might work as 
"scp -r  anonymous@ftp.astron.nl:/ftp/pub/anonymous/outgoing/adebahr/aipy/NGC315.mir . "
otherwise use generic ftp:
ftp ftp.astron.nl (anonymous:yourname) 
cd  outgoing/adebahr/aipy
and then you find the miriad directory NGC315.mir with the uv data and other relevant files.

**Comment by pkgw** [comment]
Thanks. I can reproduce on my Linux machine; will investigate.

**Comment by vdhulst** [comment]
great, thanks much for the quick reaction.

**Comment by pkgw** [comment]
Can you try installing a version of aipy using [this branch I just made, pkgw/fix-big-data](https://github.com/pkgw/aipy/tree/fix-big-data), and see if that fixes the problem? I am not sure if data sets will get read properly, but this is at least fixing the segfault in my test case.

To test that branch I believe you'll need to run the following commands:

```
git clone https://github.com/pkgw/aipy.git
cd aipy
git checkout fix-big-data
python setup.py install
```

**Comment by vdhulst** [comment]
installed the fix-big-data branch following the instructions.
problem does not occur any more as in your test case. Thanks much!!

**Comment by pkgw** [comment]
FYI, I believe that my original fix was not correct and could lead to subtle problems with mask data. Pull request #32 is, I believe, correct. Once I finish my testing I will make a new release and notify you.

**Comment by pkgw** [comment]
As far as I can tell, this is fixed in version 2.1.11. A release has been pushed to PyPI and packages on conda-forge have been updated.
